Realm/Realm/Configurations/SyncConfigurationBase.cs

```diff
@@ -103,11 +118,14 @@ internal override async Task<Realm> CreateRealmAsync(CancellationToken cancellat
 
         internal virtual Native.SyncConfiguration CreateNativeSyncConfiguration()
         {
+            var syncConfHandle = GCHandle.Alloc(this);
             return new Native.SyncConfiguration
             {
                 SyncUserHandle = User.Handle,
                 session_stop_policy = SessionStopPolicy,
                 schema_mode = Schema == null ? SchemaMode.AdditiveDiscovered : SchemaMode.AdditiveExplicit,
+                client_resync_mode = ClientResetHandler is DiscardLocalResetHandler ? ClientResyncMode.DiscardLocal : ClientResyncMode.Manual,
+                managed_sync_configuration_handle = GCHandle.ToIntPtr(syncConfHandle),
```

Comment:
Since we decided to move the session error handling in the `SyncConfigurationBase` now [SessionHandle](https://github.com/realm/realm-dotnet/blob/b300da911a10c3e66870f108caf895d8a7174ae7/Realm/Realm/Handles/SessionHandle.cs#L258) needs to get a hold of that error handler. In order not to send to native a handle per callback I sent the whole configuration, which when managed gets again knows what to do with it.
[Here](https://github.com/realm/realm-dotnet/blob/b300da911a10c3e66870f108caf895d8a7174ae7/wrappers/src/shared_realm_cs.cpp#L125) here the session error handler is called from native.
Shall I review how things are passed around? The only thing is that if there will be another callback that will be added in the future, one more will need to be passed through method parameters. I thought it was a little easier to manage in the long run.

Reply:
I was thinking we can pass the ClientResetHandler instead of the config. Then in the callback, we cast the GCHandle to `ClientResetHandlerBase` and invoke whatever methods we want on it.

Reply:
Yes I understand that. But what I was trying to say is that we also need to pass to native the [SyncConfigurationBase.OnSessionError](https://github.com/realm/realm-dotnet/blob/66db6371004aced05320eed5e5f8005765f509b1/Realm/Realm/Configurations/SyncConfigurationBase.cs#L68) in order to execute it from [SessionHandle](https://github.com/realm/realm-dotnet/blob/b300da911a10c3e66870f108caf895d8a7174ae7/Realm/Realm/Handles/SessionHandle.cs#L258).

Reply:
Ah, I understand, didn't see that. Let's discuss as part of the broader mutable struct discussion.